src/govuk/components/notification-banner/notification-banner.yaml

```diff
@@ -0,0 +1,134 @@
+- name: content
+  type: string
+  required: true
+  description: Content for the notification banner. If `contentHtml` is supplied, this is ignored.
+- name: contentHtml
+  type: string
+  required: true
+  description: HTML content for the notification banner.
+- name: type
+  type: string
+  required: false
+  description: Sets classes and attributes to create a type of notification banner. Accepts `success` or `alert`.
+- name: role
+  type: string
+  required: false
+  description: Custom role for the notification banner. Defaults to `region` (`alert` for success and error banners).
+- name: tabindex
+  type: string
+  required: false
+  description: Custom tabindex for the notification banner. Not set by default (defaults to `-1` for success and error banners).
+- name: initialFocus
+  type: string
+  required: false
+  description: Sets `data-module="govuk-initial-focus"` for the notification banner which focuses the banner on page load. Not set by default (on by default for success and error banners).
+- name: classes
+  type: string
+  required: false
+  description: Classes to add to the notification banner.
+- name: attributes
+  type: object
+  required: false
+  description: HTML attributes (for example data attributes) to add to the notification banner.
+
+examples:
+- name: default
+  data:
+    content: You have 9 days to send a response.
+- name: with content as html
+  data:
+    contentHtml: |
+      <h3 class="govuk-heading-m">This publication was withdrawn on 7 March 2014</h3><p>Archived and replaced by the <a href="#">new planning guidance</a> launched 6 March 2014 on an external website</p>
+- name: with type as success
+  data:
+    type: success
+    content: Email sent to example@email.com
+- name: with type as error
+  data:
+    type: error
+    content: There was a problem uploading your file. Please try again.
+
+# Hidden examples are not shown in the review app, but are used for tests and HTML fixtures
+
+- name: custom title
+  hidden: true
+  data:
+    title: Important information
+    content: You have 9 days to send a response.
+- name: title as html
+  hidden: true
+  data:
+    titleHtml: <span>Important information</span>
+    content: You have 9 days to send a response.
+- name: title html as text
+  hidden: true
+  data:
+    title: <span>Important information</span>
+    content: You have 9 days to send a response.
+- name: custom title heading level
+  hidden: true
+  data:
+    titleHeadingLevel: 3
+    content: You have 9 days to send a response.
+
+- name: custom content
+  hidden: true
+  data:
+    content: This publication was withdrawn on 7 March 2014.
+- name: content html as text
+  hidden: true
+  data:
+    content: <span>This publication was withdrawn on 7 March 2014.</span>
+
+- name:  custom role
+  hidden: true
+  data:
+    role: banner
+    content: You have 9 days to send a response.
+- name: custom tabindex
+  hidden: true
+  data:
+    tabindex: "0"
```

Comment:
I marked this as a string because otherwise it seems to get interpreted as a falsy value from yaml [when the template checks whether the value is set](https://github.com/alphagov/govuk-frontend/pull/1977/files#diff-8ba6864c62c415e3869d6d4bb972023bR9). We could add a Nunjucks filter that does the same as 
```
if (value or value === 0) 
```

Generally speaking, 0 not being interpreted as a truthy value for things like attributes etc. is maybe something that we should watch out for with our other yaml examples as well 🤔

Reply:
I think this would mean that users would also have to ensure they passed the `tabindex` parameter as a string?

We can't add filters as we'd then rely on those filters being loaded in our user's Nunjucks environments, but I think something like this might do what we want:

```jinja
{% if 'tabindex' in params %}
  {% set tabindex = params.tabindex %}
{% elif params.type == "success" or params.type == "error" %}
  {% set tabindex = "-1" %}
{% else %}
  {% set tabindex = false %}
{% endif %}
```

Then:

```jinja
{% if tabindex !== false %} tabindex="{{ tabindex }}" {% endif %}
```

It'd also be good to have a test that shows that the user can pass `tabindex: false` to remove the tabindex entirely.

Reply:
Thanks @36degrees this all makes sense. I'll add another test too.

Reply:
@36degrees I think I've now resolved this, please have a look.

Reply:
It'd be good to make the tabindex in the YAML file an integer again.

Reply:
Done!